frontend/src/router/guards.js

```diff
@@ -165,6 +165,18 @@ export function createGlobalBeforeGuards () {
     },
     ensureUserAuthenticatedForNonPublicRoutes(),
     ensureDataLoaded(),
+    to => {
+      const titles = []
+      titles.push(to.meta.title ?? to.name)
+      if (to.params.name) {
+        titles.push(to.params.name)
+      }
+      if (projectStore.projectName) {
+        titles.push(projectStore.projectName)
+      }
+      titles.push(process.env.VITE_APP_TITLE)
+      document.title = titles.join(' | ')
+    },
```

Comment:
<img width="633" src="https://github.com/gardener/dashboard/assets/5526658/c5439390-f66c-4956-b0a9-ce6426c7072a">
I think the title should match the breadcrumb